tests/src/rules/rename-default-import.js

```diff
@@ -0,0 +1,212 @@
+import { test } from '../utils'
+
+import { RuleTester } from 'eslint'
+
+const ruleTester = new RuleTester(),
+      rule = require('rules/rename-default-import')
+
+ruleTester.run('rename-default-import', rule, {
+  valid: [
+    test({
+      code: `import PropTypes from 'prop-types';`,
+      options: [],
+    }),
+    test({
+      code: `import PropTypes from 'prop-types';`,
+      options: [{'foo': 'Foo'}],
+    }),
+    test({
+      code: `import PropTypes from 'prop-types';`,
+      options: [{'prop-types': 'PropTypes'}],
+    }),
+    test({
+      code: `import PropTypes, {Foo} from 'prop-types';`,
+      options: [{'prop-types': 'PropTypes'}],
+    }),
+    test({
+      code: `import {default as PropTypes} from 'prop-types';`,
+      options: [{'prop-types': 'PropTypes'}],
+    }),
+    test({
+      code: `import {Foo} from 'prop-types';`,
+      options: [{'prop-types': 'PropTypes'}],
+    }),
+    test({
+      code: `import {Foo, default as PropTypes} from 'prop-types';`,
+      options: [{'prop-types': 'PropTypes'}],
+    }),
+    test({
+      code: `import * as PropTypes from 'prop-types';`,
+      options: [{'prop-types': 'PropTypes'}]
+    }),
+    test({
+      code: `const PropTypes = require('prop-types');`,
+      options: [{'prop-types': 'PropTypes'}]
+    }),
+    test({
+      code: `const object = require('prop-types').object;`,
+      options: [{'prop-types': 'PropTypes'}]
+    })
+  ],
+  invalid: [
+    test({
+      code: `import propTypes from 'prop-types';`,
+      options: [{'prop-types': 'PropTypes'}],
+      output: `import PropTypes from 'prop-types';`,
+      errors: [{
+        ruleId: 'rename-default-import',
+        message: `Default import from 'prop-types' should be aliased to PropTypes, not propTypes`,
+      }],
+    }),
+    test({
+      code: `const propTypes = require('prop-types');`,
+      options: [{'prop-types': 'PropTypes'}],
+      output: `const PropTypes = require('prop-types');`,
```

Comment:
this isn't a safe autofix - anything that referenced `propTypes` would have to be updated too.

Reply:
this is covered below:

```
test({
  code: `
  const propTypes = require('prop-types');

  const obj = {
    foo: propTypes.string
  }
  `,
  options: [{'prop-types': 'PropTypes'}],
  output: `
  const PropTypes = require('prop-types');

  const obj = {
    foo: PropTypes.string
  }
  `,
  errors: [{
    ruleId: 'rename-default-import',
    message: `Default import from 'prop-types' should be aliased to PropTypes, not propTypes`,
  }, {
    ruleId: 'rename-default-import',
    message: `Using incorrect binding name 'propTypes' instead of PropTypes for default import from package prop-types`
  }],
}),
```

Reply:
What about if someone has done `export { propTypes }` or referenced it in an object like `{ propTypes }`?

Reply:
I'd ensure the export is unfixable, the object could probably be rewritten to the long form - but I'm not sure if it wouldn't be better to also not provide autofixing there since it's probably more likely that someone wants to manually fix it whatever code is receiving the object.

Reply:
Updated. Now I don't fix the `export` statements at all.